roofit/roofitcore/src/RooPolyVar.cxx

```diff
@@ -170,21 +177,42 @@ double RooPolyVar::analyticalIntegral(Int_t code, const char* rangeName) const
   R__ASSERT(code==1) ;
 
   const double xmin = _x.min(rangeName), xmax = _x.max(rangeName);
-  const int lowestOrder = _lowestOrder;
   const unsigned sz = _coefList.getSize();
   if (!sz) return xmax - xmin;
-  _wksp.clear();
-  _wksp.reserve(sz);
-  {
-    const RooArgSet* nset = _coefList.nset();
-    unsigned i = 1 + lowestOrder;
-    for(auto * c : static_range_cast<RooAbsReal*>(_coefList)) {
-      _wksp.push_back(c->getVal(nset) / double(i));
-      ++i;
-    }
+
+  fillCoeffValues();
+
+  return RooFit::Detail::AnalyticalIntegrals::polyVarIntegral(_wksp.data(), _x.min(rangeName), _x.max(rangeName),
+                                                              _lowestOrder, sz);
+}
+
+void RooPolyVar::translate(RooFit::Detail::CodeSquashContext &ctx) const
+{
+  const unsigned sz = _coefList.size();
+
+  if (!sz) {
+    ctx.addResult(this, std::to_string((_lowestOrder ? 1. : 0.)));
+    return;
   }
-  double min = _wksp[sz - 1], max = _wksp[sz - 1];
-  for (unsigned i = sz - 1; i--; )
-    min = _wksp[i] + xmin * min, max = _wksp[i] + xmax * max;
-  return max * std::pow(xmax, 1 + lowestOrder) - min * std::pow(xmin, 1 + lowestOrder);
+
+  ctx.addResult(this, ctx.buildCall("RooFit::Detail::EvaluateFuncs::rooPolyVarEvaluate", _coefList, (double)sz,
+                                    (double)_lowestOrder, _x));
```

Comment:
```suggestion
  ctx.addResult(this, ctx.buildCall("RooFit::Detail::EvaluateFuncs::rooPolyVarEvaluate", _coefList, sz,
                                    _lowestOrder, _x));
```
I don't like casting to double here. In the generated code, this will appear then as `2.0` or `0.0`, and it gets casted back to int then. It should stay integer from the beginning.

Also, I would consider moving the `translate` function just below the `evaluate` function in the source file, so they are easier to compare and validate by eye.